internal/graph/check.go

```diff
@@ -835,33 +835,26 @@ func buildTupleKeyConditionFilter(ctx context.Context, reqCtx *structpb.Struct,
 // nolint:unused
 type usersetDetailsFunc func(*openfgav1.TupleKey) (string, string, error)
 
-// nolint:unused
-func buildUsersetDetails(typesys *typesystem.TypeSystem, t *openfgav1.TupleKey, computedRelation string) (string, string, error) {
-	object, relation := tuple.SplitObjectRelation(t.GetUser())
-	objectType, objectID := tuple.SplitObject(object)
-	cr := relation
-	if computedRelation != "" {
-		// In the case computedRelation is provided (which is the case for TTU),
-		// the computedRelation will be used. Otherwise (which is the case for
-		// userset), the computed relation will be derived from the tuple key's
-		// userset's relation.
-		cr = computedRelation
-	}
-Resolve:
-	for {
-		rel, err := typesys.GetRelation(objectType, cr)
-		if err != nil {
-			return "", "", err
-		}
-		rewrite := rel.GetRewrite()
-		switch rewrite.GetUserset().(type) {
-		case *openfgav1.Userset_ComputedUserset:
-			cr = rewrite.GetComputedUserset().GetRelation()
-		case *openfgav1.Userset_This:
-			break Resolve
-		default:
-			return "", "", fmt.Errorf("unsupported rewrite %s", rewrite.String())
-		}
+func getComputedRelation(typesys *typesystem.TypeSystem, objectType, relation string) (string, error) {
+	rel, err := typesys.GetRelation(objectType, relation)
+	if err != nil {
+		return "", err
+	}
+	rewrite := rel.GetRewrite()
+	switch rewrite.GetUserset().(type) {
+	case *openfgav1.Userset_ComputedUserset:
+		return getComputedRelation(typesys, objectType, rewrite.GetComputedUserset().GetRelation())
+	case *openfgav1.Userset_This:
+		return relation, nil
+	default:
+		return "", fmt.Errorf("unsupported rewrite %s", rewrite.String())
+	}
+}
+
+func buildUsersetDetails(typesys *typesystem.TypeSystem, objectType, objectID, relation string) (string, string, error) {
```

Comment:
**(question)**: It doesn't seem that `objectID` is used in any way in this func, we pass it in the parameters and then return it in case of no error. Am I missing anything or can we simply remove it both from input and output?

Reply:
Good observation, we can clean this up